### PR TITLE
WIP: Allow configuring http client timeout

### PR DESCRIPTION
**Pull request by sabrehagen**
## Which problem is this PR solving?

- Users are unable to configure http client timeout

## Short description of the changes

- Added support for `JAEGER_REPORTER_TIMEOUT` environment variable. Updated docs. Added tests.

## Status

Not ready for merge. Awaiting `configuration_env.js` test.

### CODE REVIEW

File: src/reporters/http_sender.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ export default class HTTPSender {
     this._url = URL.parse(options.endpoint);
     this._username = options.username;
     this._password = options.password;
-    this._timeoutMS = options.timeoutMS || DEFAULT_TIMEOUT_MS;
+    this._timeoutMs = options.timeoutMs || DEFAULT_TIMEOUT_MS;
```
Comment (yurishkuro) [inline src/reporters/http_sender.js]: let's keep this backwards compatible: `options.timeoutMs || options.timeoutMS || DEFAULT_TIMEOUT_MS;`

File: src/configuration_env.js [hunk 2]
```diff
@@ -21,18 +21,20 @@ const deprecatedEnvVars = {
...
-        if (process.env[env]) {
-          console.warn(`You are using deprecated env variable ${env}. Use ${deprecatedEnvVars[env]} instead. \nDeprecated env variable will be removed in the next major release (4.x.x)`)
-        }
-      })
+    Object.keys(deprecatedEnvVars).forEach(env => {
+      if (process.env[env]) {
+        console.warn(
+          `You are using deprecated env variable ${env}. Use ${
+            deprecatedEnvVars[env]
+          } instead. \nDeprecated env variable will be removed in the next major release (4.x.x)`
```
Comment (yurishkuro) [inline src/configuration_env.js]: This feels like a very weird formatting of an extrapolated expression. Why do we need this change in this PR?

File: src/configuration.js [hunk 3]
```diff
@@ -143,6 +144,10 @@ export default class Configuration {
       if (config.reporter.agentPort) {
         senderConfig['port'] = config.reporter.agentPort;
       }
+
+      if (config.reporter.timeout) {
```
Comment (yurishkuro) [inline src/configuration.js]: timeout or timeoutMs? Is there a test?

File: src/reporters/http_sender.js (hunk 1 above)
Comment (sabrehagen) [reply src/reporters/http_sender.js]: Sure, can do. `timeoutMS` was not referenced elsewhere in this codebase, so I felt safe changing it. Is it possible that it's used by code outside the codebase that passes `timeoutMS`?

Comment (yurishkuro) [reply src/reporters/http_sender.js]: http sender is not hidden under some `internal` package, so technically it's a public API, and I'd prefer to not break anyone who may be wiring components manually.

File: src/configuration_env.js (hunk 2 above)
Comment (sabrehagen) [reply src/configuration_env.js]: Sorry for the confusion, I should have commented this change. This is happening when `prettier` [lints](https://github.com/jaegertracing/jaeger-client-node/blob/master/package.json#L81) the staged files. How would you like this to be resolved? Alter prettier config or accept new styling?

File: src/reporters/http_sender.js (hunk 1 above)
Comment (sabrehagen) [reply src/reporters/http_sender.js]: Thanks for the info. Will update as requested.

File: src/configuration_env.js [hunk 4]
```diff
@@ -185,9 +196,10 @@ export default class ConfigurationEnv {
    * @param {Object} options - options, see Configuration.initTracer
    */
   static initTracer(config = {}, options = {}) {
-    ConfigurationEnv._validateEnv()
+    ConfigurationEnv._validateEnv();
```
Comment (sabrehagen) [inline src/configuration_env.js]: These 3 lines are changed automatically by prettier on `git add`.

File: src/configuration.js (hunk 3 above)
Comment (sabrehagen) [reply src/configuration.js]: Good catch. Only `configuration_env.js` was tested. Added a test for `configuration.js` [here](https://github.com/jaegertracing/jaeger-client-node/pull/350/files#diff-0563bbe36f9d317234592eb362aa4115R196).

File: src/configuration_env.js (hunk 2 above)
Comment (sabrehagen) [reply src/configuration_env.js]: To have a clean diff for now, I have overridden prettier and reverted these changes. Awaiting your final advice on the issue.

**Comment by yurishkuro** [comment]
```
1 failing
  1) initTracerFromENV should respect http reporter options from mismatching env:
     AssertionError: expected undefined to equal 225
      at Context.<anonymous> (dist/test/init_tracer.js:533:18)
```

**Comment by yurishkuro** [comment]
replaced by #465

**Pull request closed without merging**